plugins/inputs/system/memory.go

```diff
@@ -1,14 +1,21 @@
 package system
 
 import (
+	"bufio"
 	"fmt"
+	"os"
+	"strconv"
+	"strings"
 
 	"github.com/influxdata/telegraf"
 	"github.com/influxdata/telegraf/plugins/inputs"
 )
 
+var linuxMeminfoPath = "/proc/meminfo" // so we can override it in tests
+
 type MemStats struct {
-	ps PS
+	ps           PS
+	LinuxMeminfo bool
```

Comment:
ps is already reading the data from /proc/meminfo, i don't think this boolean is needed, just substitute ps for your implementation, or is there any other reason that i'm missing?